tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsUnitTest.java

```diff
@@ -0,0 +1,787 @@
+                asScala(reassignments));
+
+            assertEquals(1, reassignmentResult.size());
+            assertEquals(UnknownTopicOrPartitionException.class, reassignmentResult.get(new TopicPartition("quux", 0)).get().getClass());
+
+            Map<TopicPartition, ReassignPartitionsCommand.PartitionReassignmentState> expStates = new HashMap<>();
+
+            expStates.put(new TopicPartition("foo", 0),
+                new ReassignPartitionsCommand.PartitionReassignmentState(seq(0, 1, 2), seq(0, 1, 3), false));
+            expStates.put(new TopicPartition("foo", 1),
+                new ReassignPartitionsCommand.PartitionReassignmentState(seq(1, 2, 3), seq(1, 2, 3), true));
+
+            Tuple2<scala.collection.Map<TopicPartition, ReassignPartitionsCommand.PartitionReassignmentState>, Object> actual =
+                findPartitionReassignmentStates(adminClient, seq(
+                    new Tuple2<>(new TopicPartition("foo", 0), seq(0, 1, 3)),
+                    new Tuple2<>(new TopicPartition("foo", 1), seq(1, 2, 3))
+                ));
+
+            assertEquals(asScala(expStates), actual._1);
+            assertEquals(true, actual._2);
+
+            // Cancel the reassignment and test findPartitionReassignmentStates again.
+            scala.collection.Map<TopicPartition, Throwable> cancelResult = cancelPartitionReassignments(adminClient,
+                set(new TopicPartition("foo", 0), new TopicPartition("quux", 2)));
+
+            assertEquals(1, cancelResult.size());
+            assertEquals(UnknownTopicOrPartitionException.class, cancelResult.get(new TopicPartition("quux", 2)).get().getClass());
+
+            expStates.clear();
+
+            expStates.put(new TopicPartition("foo", 0),
+                new ReassignPartitionsCommand.PartitionReassignmentState(seq(0, 1, 2), seq(0, 1, 3), true));
+            expStates.put(new TopicPartition("foo", 1),
+                new ReassignPartitionsCommand.PartitionReassignmentState(seq(1, 2, 3), seq(1, 2, 3), true));
+
+            actual = findPartitionReassignmentStates(adminClient, seq(
+                new Tuple2<>(new TopicPartition("foo", 0), seq(0, 1, 3)),
+                new Tuple2<>(new TopicPartition("foo", 1), seq(1, 2, 3))
+            ));
+
+            assertEquals(asScala(expStates), actual._1);
+            assertEquals(false, actual._2);
+        }
+    }
+
+    @Test
+    public void testFindLogDirMoveStates() throws Exception {
+        try (MockAdminClient adminClient = new MockAdminClient.Builder().
+                numBrokers(4).
+                brokerLogDirs(Arrays.asList(
+                    Arrays.asList("/tmp/kafka-logs0", "/tmp/kafka-logs1"),
+                    Arrays.asList("/tmp/kafka-logs0", "/tmp/kafka-logs1"),
+                    Arrays.asList("/tmp/kafka-logs0", "/tmp/kafka-logs1"),
+                    Arrays.asList("/tmp/kafka-logs0", null))).
+                build()) {
```

Comment:
Minor style issue. Move the dot to the next line with method name. I see that original scala-test had this format, but it looks like style issue.
```suggestion
                    Arrays.asList("/tmp/kafka-logs0", null)))
                .build()) {
```

Reply:
fixed